pkg/vulnerabilityreport/io.go

```diff
@@ -93,9 +96,68 @@ func (r *readWriter) FindByOwner(ctx context.Context, owner kube.Object) ([]v1al
 	if err != nil {
 		return nil, err
 	}
+	// no reports found for provided owner, look for reports in related replicaset
+	if len(list.Items) == 0 && (owner.Kind == kube.KindDeployment || owner.Kind == kube.KindPod) {
+		rsName, found := r.findRelatedReplicaset(ctx, owner)
+		if !found {
+			return nil, nil
+		}
+
+		list, err = r.starboardClientset.AquasecurityV1alpha1().VulnerabilityReports(owner.Namespace).List(ctx, metav1.ListOptions{
+			LabelSelector: labels.Set{
+				kube.LabelResourceKind:      string(kube.KindReplicaSet),
+				kube.LabelResourceName:      rsName,
+				kube.LabelResourceNamespace: owner.Namespace,
+			}.String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return list.DeepCopy().Items, nil
 }
 
+// findRelatedReplicaset attempts to find the replicaset that is associated with
+// the given owner. If the owner is a Deployment, it will look for a replicaset
+// that is controlled by the Deployment. If the owner is a Pod, it will look for
+// the replicaset that owns the Pod.
+func (r *readWriter) findRelatedReplicaset(ctx context.Context, owner kube.Object) (rsName string, found bool) {
```

Comment:
I think we'll need a very similar logic to resolve ConfigAuditReports. So I was wondering if this method can be moved out the vulnerabilityreport package and made public.

Reply:
SG. Done.